### PR TITLE
feat: Add parent beacon block root into `ExecutionPayloadEnvelopeV3`

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds the `parentBeaconBlockRoot` field to the `ExecutionPayloadEnvelopeV3`, required in the `Ecotone` hardfork. 

Reference: https://github.com/ethereum-optimism/op-geth/blob/70103aa6866c3d965f1f71154958cd9c1a7beb5b/beacon/engine/types.go#L102-L103

### CODE REVIEW

File: crates/rpc-engine-types/src/payload.rs [hunk 1]
```diff
@@ -117,6 +117,9 @@ pub struct ExecutionPayloadEnvelopeV3 {
     /// Introduced in V3, this represents a suggestion from the execution layer if the payload
     /// should be used instead of an externally provided one.
     pub should_override_builder: bool,
+    /// Optimism-specific - Ecotone parent beacon block root
+    #[serde(skip_serializing_if = "Option::is_none")]
```
Comment (DaniPopes) [inline crates/rpc-engine-types/src/payload.rs:121]:
```suggestion
    #[serde(default, skip_serializing_if = "Option::is_none")]
```

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by DaniPopes — APPROVED** [review]

**Review by Evalir — APPROVED** [review]

File: crates/rpc-engine-types/src/payload.rs [hunk 2]
```diff
@@ -117,6 +117,9 @@ pub struct ExecutionPayloadEnvelopeV3 {
     /// Introduced in V3, this represents a suggestion from the execution layer if the payload
     /// should be used instead of an externally provided one.
     pub should_override_builder: bool,
+    /// Optimism-specific - Ecotone parent beacon block root
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub parent_beacon_block_root: Option<B256>,
```
Comment (mattsse) [inline crates/rpc-engine-types/src/payload.rs:122]: I'm fine with that for now, if you create a new reth issue for better rpc execution payload customization

cc @rjected

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
@clabby to update reth

please bump pinned alloy commit in paradigm/emv-inspectors first, then bump both on reth

File: crates/rpc-engine-types/src/payload.rs (hunk 2 above)
Comment (gakonst) [reply crates/rpc-engine-types/src/payload.rs:122]: is this fixed with the customized engine types in node builder?

Comment (mattsse) [reply crates/rpc-engine-types/src/payload.rs:122]: not yet https://github.com/paradigmxyz/reth/pull/6871

**Pull request merged**